aws-sdk-core/spec/aws/cloud_front/url_signer_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'spec_helper'
+
+module Aws
+  module CloudFront
+        describe UrlSigner do
+
+            let(:key_pair_id) { "CF_KEYPAIR_ID" }
+            let(:private_key_path) { "#{File.dirname(__FILE__)}/cf_private_key.pem" }
+            let(:signer) { UrlSigner.new(key_pair_id, private_key_path) }
+        	let(:expires) { 1357034400 } # January 1, 2013 10:00 am UTC (Unix timestamp)
+
+            describe '#initialize' do
+                it 'takes keypair id and privatekey path' do
+                    expect(signer.class).to eq(UrlSigner)
```

Comment:
Inline your major assertion.

```ruby
signer = UrlSigner.new(key_pair_id, private_key_path)
expect(signer).to ...
```

Try to not hide the main code under test in a let expression.